src/hooks/useCart/useCart.tsx

```diff
@@ -133,7 +133,9 @@ export const useCart = (
   useEffect(() => {
     if (
       cartQuota &&
-      (!prevCartQuota || prevIds !== ids || prevProducts !== products)
+      (prevCartQuota != cartQuota ||
```

Comment:
hmm curious, do you think we should keep the original `!prevCartQuota` and add this new line of logic?

Would there be issues if we keep only `prevCartQuota != cartQuota` and 1st quota is not retrievable upon initialisation 🤔

Reply:
I don't think there will be a case where `cartQuota` will be `undefined` for that condition to be `true`, since if it fails to fetch the quota properly, quota errors will be thrown. However, I do think that having the condition of `!prevCartQuota` reduces all ambiguity, and is a better safeguard. Good catch!

The conditions for updating the cart should be:

1. An incoming cart quota exists, AND
2. There is no existing cart quota, OR
3. The incoming cart quota is different from the existing cart quota, OR
4. The customer IDs have changed, OR
5. The products in the cart have changed

Adding the condition back should make the above clearer.

Reply:
sounds good~ :D 

small nit, shall we update the conditions in the comment above also? 😆